homeassistant/components/uvc/camera.py

```diff
@@ -1,103 +1,90 @@
 """Support for Ubiquiti's UVC cameras."""
 import logging
-import socket
+from typing import Optional
 
 import requests
-from uvcclient import camera as uvc_camera, nvr
 import voluptuous as vol
 
 from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
-from homeassistant.const import CONF_PORT, CONF_SSL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import PlatformNotReady
 import homeassistant.helpers.config_validation as cv
 
-_LOGGER = logging.getLogger(__name__)
-
-CONF_NVR = "nvr"
-CONF_KEY = "key"
-CONF_PASSWORD = "password"
+from . import CONF_KEY, CONF_NVR, CONF_PORT, CONF_SSL, DEFAULT_PORT, DEFAULT_SSL
 
-DEFAULT_PASSWORD = "ubnt"
-DEFAULT_PORT = 7080
-DEFAULT_SSL = False
+_LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_NVR): cv.string,
         vol.Required(CONF_KEY): cv.string,
-        vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
         vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
     }
 )
 
 
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Set up camera entry."""
+    return setup_platform(hass, entry.data, async_add_entities)
```

Comment:
We aren't allowed to use the sync api inside async context. We're inside the event loop here and that must never block.